File: backend/src/main/java/com/koval/trainingplannerbackend/training/zone/ZoneSystem.java
package com.koval.trainingplannerbackend.training.zone;

import com.koval.trainingplannerbackend.training.model.SportType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "zone_systems")
public class ZoneSystem {
    @Id
    private String id;
    @Indexed
    private String coachId;
    private String name;
    private SportType sportType = SportType.CYCLING;
    private ZoneReferenceType referenceType;
    private String referenceName;
    private List<Zone> zones = new ArrayList<>();
    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();
}

File: android/app/src/main/java/com/koval/trainingplanner/domain/model/ZoneSystem.kt
package com.koval.trainingplanner.domain.model

data class ZoneSystem(
    val id: String,
    val name: String,
    val sportType: SportType,
    val referenceType: String,
    val referenceName: String?,
    val referenceUnit: String?,
    val zones: List<Zone>,
    val defaultForSport: Boolean,
)

data class Zone(
    val label: String,
    val low: Int,
    val high: Int,
    val description: String?,
)
